query-algebrizer/src/types.rs

```diff
@@ -35,6 +36,14 @@ pub enum DatomsTable {
     FulltextValues,     // The virtual table mapping IDs to strings.
     FulltextDatoms,     // The fulltext-datoms view.
     AllDatoms,          // Fulltext and non-fulltext datoms.
+    Computed(usize),    // A computed table, tracked elsewhere in the query.
+}
+
+/// A source of rows.
+pub enum ComputedTable{
+    // TODO: how do we represent projecting type columns?
+    // Subquery(BTreeSet<Variable>, ::clauses::ConjoiningClauses),
+    Union(BTreeSet<Variable>, Vec<::clauses::ConjoiningClauses>),
```

Comment:
nit: Do you want a bit for `UnionDistinct` here?  You probably want to make some statements about preconditions.  I don't see exactly what you're representing, but that'll probably be clear from later context.